examples/scene/prefab.rs

```diff
@@ -0,0 +1,232 @@
+                                .ok_or_else(|| SceneSpawnError::UnregisteredType {
+                                    type_name: component.type_name().to_string(),
+                                })
+                                .unwrap();
+                            let reflect_component = registration
+                                .data::<ReflectComponent>()
+                                .ok_or_else(|| SceneSpawnError::UnregisteredComponent {
+                                    type_name: component.type_name().to_string(),
+                                })
+                                .unwrap();
+                            if world
+                                .entity(new_entity)
+                                .contains_type_id(registration.type_id())
+                            {
+                                if entry.overwrite_existing_components {
+                                    reflect_component.apply_component(
+                                        world,
+                                        new_entity,
+                                        &**component,
+                                    );
+                                }
+                            } else {
+                                reflect_component.add_component(world, new_entity, &**component);
+                            }
+                        }
+                    }
+
+                    // This is where we map the scene entities to the world entities in any component
+                    // that implements MapEntities.
+                    for registration in type_registry.iter() {
+                        if let Some(map_entities_reflect) =
+                            registration.data::<ReflectMapEntities>()
+                        {
+                            map_entities_reflect
+                                .map_entities(world, &entity_map)
+                                .unwrap();
+                        }
+                    }
+
+                    complete.push(entity);
+                }
+            }
+        });
+
+        // Remove the entity from the queue.
+        for entity in complete {
+            factory.queue.remove(&entity);
+        }
+    })
+}
+
+fn write_scene(world: &mut World) {
+    let mut scene_world = World::new();
+
+    let path = "assets/my_scene/path.scn.ron";
+
+    scene_world
+        .spawn()
+        .insert(A {
+            message: "I came from the scene file!".to_string(),
+        })
+        .insert(B {
+            data: vec![5, 7, 8, 22, 42, 1001],
+        });
+
+    let type_registry = world.get_resource::<TypeRegistry>().unwrap();
+    let scene = DynamicScene::from_world(&scene_world, type_registry);
+
+    let mut file = match File::create(path) {
+        Err(reason) => panic!("Failed to create file: {}", reason),
+        Ok(file) => file,
+    };
+
+    match file.write_all(scene.serialize_ron(type_registry).unwrap().as_bytes()) {
```

Comment:
Writing directly to a file is not recommended in a system as there is no guarantee on how long it will take.

It may be "ok" in this case as it's the setup needed to show the example, not the example itself, but then a disclaimer would be nice to recommend to not do that in actual code

Reply:
How about "It's recommended that you write the scene before calling on it, as it takes time to write the scene to a file.
// You might try to load the scene, but find that it's not written yet!"